### ISSUE TITLE
[Discussion] Make has_lengthscale an attr rather than kwarg

### ISSUE THREAD

**Issue by jahall** [cornellius-gp/gpytorch#922]
Currently `has_lengthscale` is passed in as an argument to `kernels.Kernel`. Would this not make more sense as a class attribute which can then be overridden by child classes? I can't think of situation in which it would make sense to allow a user to specify this via a kwarg...

Example proposed implementation:
```
class Kernel(...):
    _has_lengthscale = False

class RBFKernel(Kernel):
    _has_lengthscale = True
```

**Comment by jacobrgardner** [comment]
Yeah, I think this is a good idea. At the moment, a lot of the Kernel kwargs have turned in to a complete mess to be honest. Most of them are absorbed in to `**kwargs` in a lot of the subclasses, and there are a huge number of them, so converting some of them to attributes and cleaning up the `kwargs` sounds like a good thing to do for the next release.

**Comment by jahall** [comment]
Great. Also, I'm not sure it makes sense to have lengthscale support baked in to the base `Kernel`. Maybe in some `StationaryKernel` subclass or something?

**Comment by gpleiss** [comment]
@jahall - we thought about creating a StationaryKernel subclass a while back. Since most of our kernels are stationary we thought it was maybe a bit too much overhead. However, I wouldn't be opposed to adding it into the library.

**Comment by jahall** [comment]
I think it might be a helpful distinction - though I understand your reticence to introduce unnecessary class hierarchy.

**Comment by KeAWang** [comment]
Once we remove it from the kwargs, we can just change `_has_lengthscale` to `_is_stationary` which is a bit cleaner than unnecessarily subclassing 

Also I think the Toeplitz stuff right now doesn't check to see if the kernel is stationary or not.

**Comment by jahall** [comment]
I think `has_lengthscale` and `is_stationary` should be separate attributes because, for example, the product of two stationary kernels would be stationary...but would not have a lengthscale

**Comment by gpleiss** [comment]
Closed by #925
